Solutions

ingridstoleru opened the pull request:
Acest pull request conține soluții pentru următoarele exerciții:
- [x] caesar
- [ ] grep
  - [ ] advanced
  - [ ] simple
- [x] icao
  - [x] to_icao
  - [x] from_icao
- [x] paint
  - [x] cursor
  - [x] fill
- [x] paranteze
- [ ] reminder
- [x] unic
- [x] extra
  - [x] parcurgere directoare
  - [x] tree

Commit:
Adaugă exercițiul reminder

Commit:
Fixed typos and updated a requirement

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>

Commit:
Modifică structura cavoului de cod

Motivație: Noi suntem români :)

Commit:
Adaugă descriere pentru fiecare exercițiu

Commit:
Adaugă suport pentru integrare continuă

Commit:
Fix text.txt

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>

alexcoman commented:
Modificările din https://github.com/IngridSt/python-lab/commit/41d99426c1f7b79987ed80ab3258415a889d0c16 ar trebui incluse în https://github.com/IngridSt/python-lab/commit/4b76d8b9f46f0bd2767b29a0d757ce39b3604469.

alexandrucoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+
```
Linia aceasta nu este necesară.

IngridSt replied:
Cum includ modificarile in IngridSt@.... ?

IngridSt replied:
Dar Travis nu impune un newline dupa finalizarea codului?

alexcoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+
+"""Solutia problemei Caesar"""
+def decripteaza(mesaj):
+    """Identificarea cheii si decriptarea"""
```
Care este secvența de cod care identifică cheia ?

alexcoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+"""Solutia problemei Caesar"""
+def decripteaza(mesaj):
+    """Identificarea cheii si decriptarea"""
+    script= mesaj
```
Linia aceasta este inutilă.

alexandrucoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+    length= len(mesaj)
+    for key in range(26):
+        for letter in range(length):
+            if mesaj[letter]> 65 and mesaj[letter]< 97:
```
Pe setul nostru de date if-ul acesta nu se executa niciodată.

alexcoman replied:
If-ul acesta nu se va executa niciodată.

alexcoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+    for key in range(26):
+        for letter in range(length):
+            if mesaj[letter]> 65 and mesaj[letter]< 97:
+                script[letter]= script[(letter+key)%len(script)]
```
Linia aceasta va cauza o excepție.

alexcoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,27 @@
+        return
+
+    for mesaj in mesaje.splitlines():
+        decripteaza(mesaj)
```
Funcția ta va returna valoarea pe care o primește ca argument.

ingridstoleru replied:
Am dat commit la o varianta gresita a problemei. Remediez imediat.

alexcoman commented on solutii/IngridS/Cursor.py:
```diff
@@ -0,0 +1,36 @@
+
+
+def distanta(string, pozitie):
+	directie, valoare=string.split()
```
Ar trebui tratat cazul în care datele sunt invalide.

alexcoman commented on solutii/IngridS/Cursor.py:
```diff
@@ -0,0 +1,36 @@
+		print("Nu am putut obține coordonatele.")
+		return
+	pozitie=[0,0]
+	for i in mesaje.splitlines():
```
În loc de `i``ai putea folosi ceva mai sugestiv (de exemplu:`linie`).

alexcoman commented on solutii/IngridS/Cursor.py:
```diff
@@ -0,0 +1,36 @@
+	print rezultat
+
+if __name__ == "__main__":
+	main()
```
Trebuie o linie nouă la sfârșit de fișier.

alexcoman commented on solutii/IngridS/FROM_ICAO.py:
```diff
@@ -0,0 +1,47 @@
+}
+
+def inversare_dictionar():
+	d=dict()
```
Folosește nume mai sugestive.

alexcoman commented on solutii/IngridS/Fill.py:
```diff
@@ -0,0 +1,37 @@
+
```
Șterge această linie.

alexcoman commented on solutii/IngridS/Iterating.py:
```diff
@@ -0,0 +1,15 @@
+        path= os.path.join(path_to_dir, filename)
+        if os.path.isfile(path):
+            if "a" in filename:
+                print(filename)
```
Ar trebui afișată calea completă a fișierului.

alexcoman commented on solutii/IngridS/Iterating.py:
```diff
@@ -0,0 +1,15 @@
+        if os.path.isdir(path):
+            iteration(path)                
+
+iteration('C:\\Users\\Ingrid\\Desktop\\test')
```
Nu este ok.
Trebuie o linie nouă.

alexcoman commented on solutii/IngridS/Represent_as_a_tree.py:
```diff
@@ -0,0 +1,15 @@
+def Represent_as_tree(path_to_dir, depth=0):
+	files_in_dir=os.listdir(path_to_dir)
+	for filename in files_in_dir:
+		print ('\t' * depth, filename)
```
Nu trebuie spațiu între `print` și `(`.

alexcoman commented on solutii/IngridS/Represent_as_a_tree.py:
```diff
@@ -0,0 +1,15 @@
+		
+
+
+Represent_as_tree('C:\\Users\\Ingrid\\Desktop\\test')
```
Nu mai folosi path-uri hard coded.

alexcoman commented on solutii/IngridS/unic.py:
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python
+# *-* coding: UTF-8 *-*
+
+istoric=[1,2,3,4,5,1,2,3,4]
```
Nu este necesar.

alexcoman commented on solutii/IngridS/unic.py:
```diff
@@ -0,0 +1,16 @@
+        return p
+
+if __name__ == "__main__":
+        print(gaseste(istoric))
```
Poți oferi lista de mai sus direct ca argument.

alexcoman commented on solutii/IngridS/Cursor.py:
```diff
@@ -0,0 +1,36 @@
+coordonate={"stanga": [-1,0], "dreapta":[1,0], "jos":[0,-1], "sus":[0,1]}
+
+
+
```
Prea multe linii goale.

ingridstoleru committed:
Am adaugat solutia problemei Fill

Am adaugat solutia problemei Fill

Am adaugat solutia corecta a problemei Fill

Am adaugat solutia corecta a problemei Fill

Am adaugat solutia corecta a problemei FIll

ingridstoleru committed:
Am adaugat solutia problemei Iterarea printr-un director

Am adaugat solutia problemei Iterarea printr-un director

ingridstoleru committed:
Am adaugat solutia problemei Paranteze

Am adaugat solutia problemei Paranteze

Am adaugat solutia corectata a problemei Paranteze

Am adaugat solutia corecta a problemei Paranteze.py

ingridstoleru committed:
Am adaugat solutia problemei Reprezentarea ca arbore a unui director

Am adaugat solutia problemei Reprezentarea ca arbore a unui director

alexcoman commented on solutii/IngridS/Caesar.py:
```diff
@@ -0,0 +1,32 @@
+                else:
+                    caracter = chr(122-(97-aux))
+            else:
+                pass
```
Ramura `else` nu face nimic, așa că mai bine nu o pui deloc.

alexcoman commented on solutii/IngridS/FROM_ICAO.py:
```diff
@@ -0,0 +1,47 @@
+if __name__== "__main__":
+	main()	
+
+
```
Prea multe linii goale la sfârșitul fișierului.

alexcoman commented on solutii/IngridS/Paranteze.py:
```diff
@@ -0,0 +1,30 @@
+        else:
+            if caracter == ")":
+                if not Stiva or Stiva.pop() == "[":
+                    print ("Nu este corect")
```
Ai un spațiu în plus între `print` și `(`.

ingridstoleru committed:
Am adaugat solutia problemei Caesar

Am adaugat solutia problemei Caesar

Am adaugat solutia corectata a problemei Caesar

Am adaugat solutia corecta a problemei Caesar

Am adaugat solutia corecta a problemei Caesar

alexcoman commented on solutii/IngridS/Fill.py:
```diff
@@ -0,0 +1,37 @@
+"""Rezolvarea problemei Fill"""
+#!/usr/bin/env python
```
Pentru a avea efect trebuie să fie prima linie din fișier.

alexcoman commented on solutii/IngridS/Fill.py:
```diff
@@ -0,0 +1,37 @@
+"""Rezolvarea problemei Fill"""
+#!/usr/bin/env python
+# *-* coding: UTF-8 *-*
```
Dacă nu ai caractere speciale în cod, nu este necesară.

alexcoman commented on solutii/IngridS/Paranteze.py:
```diff
@@ -0,0 +1,30 @@
+
+if __name__ == "__main__":
+    EXEMPLU = "[(())]"
+    print este_corect(EXEMPLU)
```
Poate fi simplificat în:

```
este_corect("[()()]")
```

alexcoman commented on solutii/IngridS/ProblemaCaesar.py:
```diff
@@ -0,0 +1,31 @@
+        return
+    for mesaj in mesaje.splitlines():
+        decripteaza(mesaj)
+if __name__ == "__main__":
```
Lasă două linii libere după funcția main.

ingridstoleru committed:
Am adaugat solutia problemei Cursor

Am adaugat solutia corecta a problemei Cursor

ingridstoleru committed:
Am adaugat solutia problemei Unic

Am adaugat solutia corecta a problemei Unic

Am adaugat solutia corecta a problemei Unic

ingridstoleru committed:
Am adaugat solutia problemei TO_ICAO

Am adaugat solutia problemei To_Icao

Am adaugat solutia problemei FROM_ICAO

Am adaugat solutia corecta a problemei FromIcao

Am adaugat solutia corecta a problemei To_Icao

ingridstoleru committed:
mesaj neimportant

ingridstoleru committed:
Revert "mesaj neimportant"

This reverts commit 0a27f44032a015d67fdf0df223ccf4e220781cbc.
